Update Python performance to include configuration (#2594)
getsentry/sentry-docs: src/platforms/python/common/performance/index.mdx

description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
---

Sentry allows you to monitor the performance of your application, showing you how latency in one service may affect another service, and letting you pinpoint exactly which parts of a given operation may be responsible. To do this, it captures distributed traces consisting of transactions and spans, which measure individual services and individual operations within those services, respectively. You can learn more about this model in our [Distributed Tracing](/product/performance/distributed-tracing/) docs. Performance Monitoring is available for the Sentry Python SDK version ≥ 0.11.2.

## Enabling Tracing

To get started automatically sending traces you can set the `traces_sample_rate` in your SDK config. This should be a number between `0` and `1`. (For example, to send 20% of transactions, set traces_sample_rate to `0.2`.)

> <PlatformContent includePath="getting-started-config" />

Reviewer: Two things here:

1) While this is true, now that there's the `traces_sampler`, perhaps you could include more of what's here in the above? https://github.com/getsentry/sentry-docs/blob/4d159bbbbf325dde724c6f4a3152c020b0374cee/src/platforms/javascript/common/performance/index.mdx#L38-L85

2) Relatedly, I don't think this is the include we want. Can you please use this instead? `<PlatformContent includePath="performance/enable-tracing" />`. You'll need to create the `enable-tracing` folder in `includes/performance` (it exists already, but only on my branch), so that way once my work is done, your include will already be in the right place?

Author: @lobsterkatie I don't see `enable-tracing` in any other PR. Can you link me to it? What content should go in here?

Reviewer: No PR yet - just in a branch. Might be easier for me to just pull your branch and add what I'm talking about. Would you mind?

Reviewer: (To answer your question, though, a Python version of the code above.)

Reviewer: No PR yet - just in a branch. To answer your question, though, a Python version of the code above.

Basically this:

```
` ` `python
import sentry_sdk

def traces_sampler(sampling context):
    # ...

sentry_sdk.init(
  dsn="___PUBLIC_DSN___",

  # To set a uniform sample rate
  traces_sample_rate=0.2,

  # Alternatively, to control sampling dynamically
  traces_sampler=traces_sampler
)
` ` `
```

except remove the spaces from in between the backticks. I had to put them there to make them render and not get used by the GH MD engine.

## Enabling Tracing

To get started automatically sending traces you can:
 - Set a uniform sample rate for all transactions, by setting the `traces_sample_rate` option in your SDK config to a number between `0` and `1`. (For example, to send 20% of transactions, set `traces_sample_rate` to `0.2`.)
 - Control the sample rate dynamically, based on the transaction itself and the context in which it's captured, by providing a function to the `traces_sampler` config option.

<PlatformContent includePath="performance/enable-tracing" />

Your application will now start automatically capturing transactions.

<Note>

When you first enable tracing, the easiest thing to do to test it is to set `traces_sample_rate` to `1.0`, because that guarantees that every transaction will be sent to Sentry for you to see.

Once you're done with testing, however, we recommend that you either **lower your `traces_sample_rate` value, or switch to using `traces_sampler` to dynamically sample and filter your transactions**.

> Without sampling, our automatic instrumenation will send a transaction any time any user loads any page or navigates anywhere in your app. Sampling allows you to send representative data without overwhelming either your system or your Sentry transaction quota.

Reviewer: ```suggestion
Without sampling, our automatic instrumentation will send a transaction with every server request, celery task, etc (depending on which integrations you're using), which can add up quickly. Sampling allows you to send representative data without overwhelming either your system or your Sentry transaction quota.
```

description: "Learn more about how to configure our Performance integrations to get the best experience out of it."
---

Sentry allows you to monitor the performance of your application, showing you how latency in one service may affect another service, and letting you pinpoint exactly which parts of a given operation may be responsible. To do this, it captures distributed traces consisting of transactions and spans, which measure individual services and individual operations within those services, respectively. You can learn more about this model in our [Distributed Tracing](/product/performance/distributed-tracing/) docs. Performance Monitoring is available for the Sentry Python SDK version ≥ 0.11.2.

## Enabling Tracing

To get started automatically sending traces you can set the `traces_sample_rate` in your SDK config. This should be a number between `0` and `1`. (For example, to send 20% of transactions, set traces_sample_rate to `0.2`.)

> ```python
> import sentry_sdk
> 
> sentry_sdk.init(
>     "___PUBLIC_DSN___",
> 
>     # Set traces_sample_rate to 1.0 to capture 100%
>     # of transactions for performance monitoring.
>     # We recommend adjusting this value in production.
>     traces_sample_rate=1.0,
> )

Reviewer 2: You can change this to use an include
```suggestion
<PlatformContent includePath="getting-started-config" />
```

```

Your application will now start automatically capturing transactions.

> <Alert level="info" title="Note">

Reviewer 2: ```suggestion
<Note>
```

When you first enable tracing, the easiest thing to do to test it is to set `traces_sample_rate` to `1.0`, because that guarantees that every transaction will be sent to Sentry for you to see.

Once you're done with testing, however, we recommend that you either **lower your `traces_sample_rate` value, or switch to using `traces_sampler` to dynamically sample and filter your transactions**.

> Without sampling, our atomatic instrumenation will send a transaction any time any user loads any page or navigates anywhere in your app. That's a lot of transactions! Sampling allows you to get representative data without overwhelming either your system or your Sentry transaction quota.

Reviewer 2: We generally avoid exclamation points :) 
```suggestion
Without sampling, our automatic instrumenation will send a transaction any time any user loads any page or navigates anywhere in your app. Sampling allows you to send representative data without overwhelming either your system or your Sentry transaction quota.
```

Reviewer: I'd also suggest changing this to "will send a transaction with every request to your server" or some such - since it's not just pageloads and navigations that generate requests, and not just webapps that servers serve.

> </Alert>

Reviewer 2: ```suggestion
</Note>
```
